mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,714 @@
+from math import ceil, log
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import ConvModule
+
+from mmdet.core import multi_apply
+from mmdet.ops import CornerPool, soft_nms
+from ..builder import HEADS, build_loss
+from ..utils import gaussian_radius, gen_gaussian_target
+
+
+class BCPool(nn.Module):
+    """Bidirectional Corner Pooling Module (TopLeft, BottomRight, etc.)
+
+    Args:
+        in_channels (int): Input channels of module.
+        pool_direction1 (str): direction of the first CornerPool.
+        pool_direction2 (str): direction of the second CornerPool.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 pool_direction1,
+                 pool_direction2,
+                 norm_cfg=dict(type='BN', requires_grad=True)):
+        super(BCPool, self).__init__()
+        self.pool1_conv = ConvModule(
+            in_channels, 128, 3, padding=1, norm_cfg=norm_cfg)
+        self.pool2_conv = ConvModule(
+            in_channels, 128, 3, padding=1, norm_cfg=norm_cfg)
+
+        self.p_conv = ConvModule(
+            128, in_channels, 3, padding=1, norm_cfg=norm_cfg, act_cfg=None)
+
+        self.conv1 = ConvModule(
+            in_channels, in_channels, 1, norm_cfg=norm_cfg, act_cfg=None)
+        self.conv2 = ConvModule(
+            in_channels, in_channels, 3, padding=1, norm_cfg=norm_cfg)
+
+        self.pool1 = CornerPool(pool_direction1)
+        self.pool2 = CornerPool(pool_direction2)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
```

Comment:
We update docstrings in the whole codebase recently and we also require docstrings to functions like `forward`.